msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md

```diff
@@ -0,0 +1,256 @@
+|`TeamsAppInstallation.ReadWriteSelfForTeam.All`|Allows a Teams app to read, install, upgrade, and uninstall itself in any **team**, without prior sign in or use.|
+
+To use these permissions, you must add a [webApplicationInfo](../../resources/schema/manifest-schema.md#webapplicationinfo) key to your app manifest with the following values:
+> [!div class="checklist"]
+> [!div class="checklist"]
+>
+> * **id**  — your Azure AD app id.
+> * **resource** — the resource URL for the app.
+>
+
+>[!NOTE]
+>
+> * Your bot requires _application_ not _user delegated_ permissions because the installation is not for yourself but for others.
+>
+> * An Azure AD tenant administrator must [explicitly grant permissions to an application](/graph/security-authorization#grant-permissions-to-an-application). After an application is granted permissions, _all_ members of the Azure AD tenant will gain the granted permissions.
+
+## Enable proactive app installation and messaging
+
+ > [!IMPORTANT]
+> Microsoft Graph will only install apps published in your organization's [app catalog](../../concepts/deploy-and-publish/overview.md#publish-to-your-organizations-app-catalog), i.e., the tenant app catalog.
+
+### ✔ [Create](../../bots/how-to/create-a-bot-for-teams.md) and [Publish](../../concepts/deploy-and-publish/overview.md) your [proactive messaging bot](../../concepts/bots/bot-conversations/bots-conv-proactive.md) for Teams
+
+>[!TIP]
+> The production-ready [**Company Communicator**](../..//samples/app-templates.md#company-communicator) app template enables broadcast messaging and is a good foundation for building your proactive bot application.
+
+### ✔ Get the `teamsAppId` for your app
+
+You will need the   `teamsAppId`   for the next steps. It can be retrieved from your organization's app catalog.  
+
+**Microsoft Graph page reference.** [teamsApp resource type](/graph/api/resources/teamsapp?view=graph-rest-1.0)
+
+**HTTP GET** request:
+
+```http
+GET /appCatalogs/teamsApps?$filter=externalId eq '{IdFromManifest}'
+```
+
+The request will return a `teamsCatalogApp`  object. The returned object's `id`  is the app's catalog generated app ID and is different from the id that you provided in your Teams app manifest:
+
+```json
+{
+  "value": [
+    {
+      "id": "b1c5353a-7aca-41b3-830f-27d5218fe0e5",
+      "externalId": "f31b1263-ba99-435a-a679-911d24850d7c",
+      "name": "Test App",
+      "version": "1.0.1",
+      "distributionMethod": "Organization"
+    }
+  ]
+}
+```
+
+### ✔ Determine whether your bot is currently installed for a message recipient
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppId}'
```

Comment:
This can be simplified too:

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}'